spec/models/piece_spec.rb

```diff
@@ -8,4 +8,36 @@
         .to eq('white-queen.png')
     end
   end
+  describe "#move_to!" do
+    let(:king) { create(:king, x_position: 1, y_position: 1) }
+    context "when the tile is empty" do
+      it "moves to the coordinates" do
+        king.move_to!(2, 2)
+        expect(king.x_position).to eq(2)
+        expect(king.y_position).to eq(2)
+      end
+    end
+    context "when the tile is not empty" do
+      let(:white_king) { create(:king, color: true) }
+      let!(:black_knight) { create(:knight, color: false) }
```

Comment:
same comment as above: pieces are currently being created on top of existing pieces.